pvlib/iotools/sodapro.py

```diff
@@ -0,0 +1,368 @@
+    GHI, ghi†                 float   Global horizontal radiation
+    BHI, bhi†                 float   Beam (direct) radiation on horizontal
+    DHI, dhi†                 float   Diffuse horizontal radiation
+    BNI, dni†                 float   Beam (direct) radiation normal to the sun
+    Reliability†              float   Reliable data fraction in summarization
+    ========================  ======  =========================================
+
+    †Parameters only returned if identifier='cams_radiation'. For description
+    of additional output parameters in verbose mode, see [1]_ and [2]_.
+
+    Note that it is recommended to specify the latitude and longitude to at
+    least the fourth decimal place.
+
+    Variables corresponding to standard pvlib variables are renamed,
+    e.g. `sza` becomes `solar_zenith`. See the
+    `pvlib.iotools.cams.CAMS_VARIABLE_MAP` dict for the complete
+    mapping.
+
+    See Also
+    --------
+    pvlib.iotools.read_cams, pvlib.iotools.parse_cams
+
+    Raises
+    ------
+    requests.HTTPError
+        If the request is invalid, then an XML file is returned by the CAMS
+        service and the error message will be raised as an exception.
+
+    References
+    ----------
+    .. [1] `CAMS Radiation Service Info
+       <http://www.soda-pro.com/web-services/radiation/cams-radiation-service/info>`_
+    .. [2] `CAMS McClear Service Info
+       <http://www.soda-pro.com/web-services/radiation/cams-mcclear/info>`_
+    .. [3] `CAMS McClear Automatic Access
+       <http://www.soda-pro.com/help/cams-services/cams-mcclear-service/automatic-access>`_
+    """
+    try:
+        time_step_str = TIME_STEPS_MAP[time_step]
+    except KeyError:
+        raise ValueError(f'Time step not recognized. Must be one of '
+                         f'{list(TIME_STEPS_MAP.keys())}')
+
+    if (verbose is True) & ((time_step != '1min') | (time_ref != 'UT')):
+        verbose = False
+        warnings.warn("Verbose mode only supports 1 min. UT time series!")
+
+    if identifier not in ['mcclear', 'cams_radiation']:
+        raise ValueError('Identifier must be either mcclear or cams_radiation')
+
+    # Format verbose variable to the required format: {'true', 'false'}
+    verbose = str(verbose).lower()
+
+    if altitude is None:  # Let SoDa get elevation from the NASA SRTM database
+        altitude = -999
+
+    # Start and end date should be in the format: yyyy-mm-dd
+    start_date = start_date.strftime('%Y-%m-%d')
+    end_date = end_date.strftime('%Y-%m-%d')
+
+    email = email.replace('@', '%2540')  # Format email address
+    identifier = 'get_{}'.format(identifier.lower())  # Format identifier str
+
+    base_url = f"http://{server}/service/wps"
+
+    data_inputs = (
+        f"latitude={latitude};longitude={longitude};altitude={altitude};"
+        f"date_begin={start_date};date_end={end_date};time_ref={time_ref};"
+        f"summarization={time_step_str};username={email};verbose={verbose}")
```

Comment:
Perhaps worth a comment that this text all represents a single GET parameter -- I was scratching my head why these weren't just included as pairs in the `params` dict until I saw that they are delimited by semicolons instead of ampersands. 

Idea:  would this be easier to read with something like this?
```python
data_inputs_dict = {
   'latitude': latitude,
   ...
}
data_inputs = ";".join([f"{key}={value}" for key, value in data_inputs_dict.items()])
```

Reply:
Very nice solution. This makes it resemble a traditional params dictionary as much as possible and gives a nice overview.